### ISSUE TITLE
Acceptable characters for project names

### ISSUE THREAD

**Issue by pgporada** [metalseargolid/protokoll#5]
```
$ protokoll project create jenkins-build-app
Traceback (most recent call last):
  File "/usr/local/bin/protokoll", line 9, in <module>
    load_entry_point('protokoll==0.1.3', 'console_scripts', 'protokoll')()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/__main__.py", line 44, in create
    dbo.create_project(project_name, close=True)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/db.py", line 59, in create_project
    "is_running INT NOT NULL)", close, False, tn=project_name)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/db.py", line 242, in __execute
    cursor.execute(query)
sqlite3.OperationalError: near "-": syntax error
```

I then put quotes around the project name.
```
$ protokoll project create "jenkins-build-app"
Traceback (most recent call last):
  File "/usr/local/bin/protokoll", line 9, in <module>
    load_entry_point('protokoll==0.1.3', 'console_scripts', 'protokoll')()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/__main__.py", line 44, in create
    dbo.create_project(project_name, close=True)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/db.py", line 59, in create_project
    "is_running INT NOT NULL)", close, False, tn=project_name)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/db.py", line 242, in __execute
    cursor.execute(query)
sqlite3.OperationalError: near "-": syntax error
```

I then tried to escape the hyphens but got blown up again. Escaping the escapes also causes protokoll to explode.
```
$ protokoll project create "jenkins\-build\-app"
Traceback (most recent call last):
  File "/usr/local/bin/protokoll", line 9, in <module>
    load_entry_point('protokoll==0.1.3', 'console_scripts', 'protokoll')()
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 700, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 680, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 1027, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 873, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/dist-packages/click/core.py", line 508, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/__main__.py", line 44, in create
    dbo.create_project(project_name, close=True)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/db.py", line 59, in create_project
    "is_running INT NOT NULL)", close, False, tn=project_name)
  File "/usr/local/lib/python2.7/dist-packages/protokoll/db.py", line 242, in __execute
    cursor.execute(query)
sqlite3.OperationalError: unrecognized token: "\"
```

**Comment by pgporada** [comment]
Turns out that the acceptable characters for sqlite are alphanumeric and underscores only. If protokoll could automagically convert unacceptable chars to underscores, that would be nifty.

**Comment by metalseargolid** [comment]
This is because each project is a table in the database (though this will not be the case when I eventually write additional DB backends, I just wanted to get it working with SQLite), so you have those character limits.

I don't like the idea of autoconverting (as easy as it would be) but I can check the input prior to creating the project to check for illegal chars so it doesn't blow up like that.

**Comment by metalseargolid** [comment]
b16f7b7f66d8eb7903c1769bee956675a1c28e49 fixes the issue. This was inherently fixed in #8. I've still got one more bug to iron out before I can merge it into develop.
